### PR TITLE
feat: allow 3rd party domains

### PR DESCRIPTION
**Pull request by Ehesp**
Enables support of 3rd party custom domains!

Steps:

- Add the domain & repo to the `domains.txt` file:
	```
    melos.invertase.dev invertase/melos
	```
- Point your domain DNS at the proxy (@Salakar to document)

Some considerations:

- When developing locally, a hosts entry needs to be added:
	```txt
    127.0.0.1       melos.invertase.dev
	```
- Since pages are statically generated, you can only have a single instance of the pages HTML. In this case, if a domain has been added we force usage of the domain (and not docs.page), since pages will be generated assuming the domain is in use; `href` fields will be created as `/foo` rather than the default `/:org/:repo/foo`.
- Incoming requests for a domain are rewritten (see `next.config.js`), so the entire flow is the same (bar a few logic exceptions).

### CODE REVIEW

**Comment by Salakar** [comment]
#### 📝Changed routes:


| [**/\[...slug\]**](https://oss-docs-5xo7ipft6-invertase.vercel.app/[...slug]) |
|:-:|
|<a href="https://oss-docs-5xo7ipft6-invertase.vercel.app/[...slug]">Screenshot of /[...slug]</a><br /><sup><a href="https://images.weserv.nl?url=https://sjc.microlink.io/42lfjaMdgHQ0dbahZeZtVpUwjfwNBqDqKtAQkA3DJKVCimj2nJz8CkhnP9sG6bj-wMi8SowiKLs3HkNjIESQ3w.png">(view full size)</a> |







Commit 78b0678962e173f8f74d8fbe2156e036ce5fba16 (https://oss-docs-5xo7ipft6-invertase.vercel.app).

File: domains.txt [hunk 1]
```diff
@@ -1,3 +1 @@
-firebase.flutter.dev firebaseextended/flutterfire
-melos.invertase.dev invertase/melos
-ehesp.invertase.io ehesp/testing
+melos.invertase.io invertase/melos
```
Comment (Salakar) [inline domains.txt:1]: Shouldn't this be the .dev domain?

Comment (Ehesp) [reply domains.txt:1]: Yeah was for testing mainly. Couldn't actually find the .dev domain, but thought I'd leave it until the proxy is ready.

Comment (Salakar) [inline domains.txt:1]:
```suggestion
melos.invertase.dev invertase/melos
```

File: docs/custom-domains.mdx [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+A couple of manual steps are required before you can serve docs via your custom domain.
+
+## 1. Domain DNS
+
+The first step is to point your domain at our proxy service. Using your domain provider, create a
+DNS record:
+
+| invertase.dev | record type: |  value  |
+| :-----------: | :----------: | :-----: |
+|       @       |      A       | 0.0.0.0 |
```
Comment (Salakar) [inline docs/custom-domains.mdx:27]:
```suggestion
| invertase.dev |  record type  |      value      |
| :-----------: | :-----------: | :-------------: |
|       @       |     CNAME     | proxy.docs.page |
```

File: docs/custom-domains.mdx [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+
+| invertase.dev |  record type  |      value      |
+| :-----------: | :-----------: | :-------------: |
+|       @       |     CNAME     | proxy.docs.page |
+
+Subdomains are also supported:
+
+| invertase.dev | record type: |  value  |
+| :-----------: | :----------: | :-----: |
+|     melos     |      A       | 0.0.0.0 |
```
Comment (Salakar) [inline docs/custom-domains.mdx:33]:
```suggestion
| invertase.dev |  record type  |      value      |
| :-----------: | :-----------: | :-------------: |
|     melos     |     CNAME     | proxy.docs.page |
```

**Pull request merged**